cppapi/client/devapi_attr.cpp

```diff
@@ -542,46 +542,44 @@ DeviceAttribute & DeviceAttribute::operator=(DeviceAttribute &&rval)
 }
 #endif
 
+void DeviceAttribute::init_common_class_members(const char * new_name, int x_dim = 1, int y_dim = 0)
+{
+    name = new_name;
+    dim_x = x_dim;
+    dim_y = y_dim;
+    w_dim_x = 0;
+    w_dim_y = 0;
+    quality = Tango::ATTR_VALID;
+    data_format = Tango::FMT_UNKNOWN;
+    d_state_filled = false;
+    exceptions_flags.set(failed_flag);
+    exceptions_flags.set(isempty_flag);
+}
+
 //-----------------------------------------------------------------------------
 //
 // DeviceAttribute::DeviceAttribute() - constructor for short
 //
 //-----------------------------------------------------------------------------
 
-DeviceAttribute::DeviceAttribute(string &new_name, short datum):ext(new DeviceAttributeExt)
+DeviceAttribute::DeviceAttribute(string &new_name, short datum)
+    : ext(new DeviceAttributeExt)
 {
-	name = new_name;
-	dim_x = 1;
-	dim_y = 0;
-	w_dim_x = 0;
-	w_dim_y = 0;
-	quality = Tango::ATTR_VALID;
-	data_format = Tango::FMT_UNKNOWN;
-	d_state_filled = false;
-	exceptions_flags.set(failed_flag);
-	exceptions_flags.set(isempty_flag);
-	ShortSeq = new(DevVarShortArray);
-	ShortSeq->length(1);
-	ShortSeq[0] = datum;
-	data_type = DEV_SHORT;
+    init_common_class_members(new_name.c_str());
+    ShortSeq = new(DevVarShortArray);
```

Comment:
Is ShortSeq a member? then it must be renamed to *short_seq*. I think we need a holder class here:

```c++
sequence = new HolderClass(datum)
```

*HolderClass* can be templatized in this case: 

```c++
temlate<typename T>
class Holder {
T value; //actual sequence

public:
  template<typename T. typename = std::enable_if<short>>
  HolderClass(short datum){
     value = new ShortSeq(1);
     value.length(1);
     value[0] = datum;
   }
}

```
NOTE this is just a C++-like pseudo code (mostly inspired by Java generics)

Probably this task is for another PR